### DISCUSSION TITLE
HasManyRepeater: watch for value change

### DISCUSSION THREAD

**Discussion by ghost** [filamentphp/filament#942] [General]
Hello @danharrin, inside a HasManyRepeater, I'm trying to show/hide a grid of fields, based on the value of a field.

I'm using the following function but it's not working.

```
->visible(function ($state, Closure $get, Model $model) {
      return $get('type') == 'DATE'; // <---- not working
  }),
```

<img width="680" alt="Capture d’écran 2021-12-20 à 12 11 50" src="https://user-images.githubusercontent.com/5849907/146758475-a045813d-3889-4e28-a97b-0f07456f13d5.png">

The idea is to show the grid of dates only if the type is DATE.

Here is the code so far:

```
Forms\Components\HasManyRepeater::make('fields')
    ->relationship('fields')
    ->schema([
        Forms\Components\Grid::make(8)->schema([
            Forms\Components\TextInput::make('label')->required()->columnSpan(5),
            Forms\Components\Select::make('type')->options( Arr::pluck(FieldType::cases(), 'name', 'name'))->reactive()->required()->columnSpan(3)
        ]),
        Forms\Components\Grid::make()->schema([
            Forms\Components\DatePicker::make('date_minimum'),
            Forms\Components\DatePicker::make('date_maximum')
        ])->visible(function ($state, Closure $get, Model $model) {
            return $get('type') == 'DATE'; // <---- not working
        }),
        Forms\Components\Grid::make(8)->schema([
            Forms\Components\Toggle::make('required')
        ])
])->label(false)
```

**Comment by danharrin** [comment]
Hey, does it work with `===` instead of `==`? Saw an issue with that the other day.

Also, can you dd `$get('type')` and see what it returns?

**Reply by ghost** [reply]
`===` doesn't work neither, and `$get('type')` is always null.

**Reply by danharrin** [reply]
Is the `visible()` method executed correctly each time you update the select?

**Reply by ghost** [reply]
I put a notification and indeed I see a notification everytime the type column changes. So the visible is called but the get is null.

**Reply by ghost** [reply]
By the way, the visible is called if any reactive component is changed, not only the type selectbox.

**Reply by ryangjchandler** [reply]
@zenati Any reactive field changes trigger an entire re-render/state update of the form, so it'll call all of the visible/hidden callbacks.

**Reply by ghost** [reply]
@ryangjchandler Thanks, good to know.

**Reply by ghost** [reply]
@danharrin Something else strange, yesterday I was on version 2.5.31, and today I upgraded to 2.6.3. Since the upgrade, the confirmation dialog modals no longer appear. So I downgraded to 2.5.31 and they worked again. I guess there is an issue with the 2.6.3 ?

**Reply by danharrin** [reply]
> @danharrin Something else strange, yesterday I was on version 2.5.31, and today I upgraded to 2.6.3. Since the upgrade, the confirmation dialog modals no longer appear. So I downgraded to 2.5.31 and they worked again. I guess there is an issue with the 2.6.3 ?

Can you make sure that your view cache is clear?

```
php artisan view:clear
```

**Reply by ghost** [reply]
@danharrin I upgraded to 2.6.3 and view:clear but the two bugs are still there:
1. $get('type') empty
2. the confirmation dialogs not showing

Once I go back to 2.5.31, the 2. is OK, but 1. is still an issue.

**Reply by danharrin** [reply]
Okay, please open 2 separate issues with minimal reproducible code. No models or anything please :)

**Reply by jcgmalave** [reply]
I don't know if it's the most optimal but I did it by creating a session variable inside the element that receives the variable get session(['s_size_type_id' => $get('size_type_id')]);
